hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/server/TopicMetadataLoadingJob.java

```diff
@@ -0,0 +1,53 @@
+package pl.allegro.tech.hermes.frontend.server;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import pl.allegro.tech.hermes.common.config.ConfigFactory;
+import pl.allegro.tech.hermes.common.config.Configs;
+
+import javax.inject.Inject;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+
+public class TopicMetadataLoadingJob implements Runnable {
+
+    private final TopicMetadataLoadingRunner topicMetadataLoadingRunner;
+    private final ScheduledExecutorService executorService;
+    private final int intervalSeconds;
+
+    private ScheduledFuture job;
+
+    @Inject
+    public TopicMetadataLoadingJob(TopicMetadataLoadingRunner topicMetadataLoadingRunner,
+                                   ConfigFactory config) {
+        this(topicMetadataLoadingRunner, config.getIntProperty(Configs.FRONTEND_TOPIC_METADATA_REFRESH_JOB_INTERVAL_SECONDS));
+    }
+
+    TopicMetadataLoadingJob(TopicMetadataLoadingRunner topicMetadataLoadingRunner, int intervalSeconds) {
+        this.topicMetadataLoadingRunner = topicMetadataLoadingRunner;
+        this.intervalSeconds = intervalSeconds;
+
+        ThreadFactory threadFactory = new ThreadFactoryBuilder()
+                .setNameFormat("TopicMetadataLoadingJob-%d").build();
+        this.executorService = Executors.newSingleThreadScheduledExecutor(threadFactory);
+    }
+
+    @Override
+    public void run() {
+        topicMetadataLoadingRunner.refreshMetadata();
```

Comment:
just one thing here - please add `try {} catch(Exception)`, with proper logging, as if you will throw any exception from scheduled code, no more jobs will be ever scheduled. just a precaution. otherwise jobs will stop running and we will never know it occurred

Reply:
to clarify: I can see that you have error handling in `TopicMetadataLoadingRunner` class, but it is better to add this layer explicitly in `run` method, so future changes in `TopicMetadataLoadingRunner` won't introduce any new exceptions that are thrown